src/mapblock.cpp

```diff
@@ -484,8 +506,16 @@ void MapBlock::deSerialize(std::istream &is, u8 version, bool disk)
 		throw SerializationError("MapBlock::deSerialize(): invalid content_width");
 	if(params_width != 2)
 		throw SerializationError("MapBlock::deSerialize(): invalid params_width");
-	MapNode::deSerializeBulk(is, version, data, nodecount,
+
+	if (version >= 29) {
+		MapNode::deSerializeBulk(is, version, data, nodecount,
+			content_width, params_width);
+	} else {
+		// use iss from above to avoid allocating another stream object
```

Comment:
Same here. Recycling objects that are for another purpose only makes this code less readable than it already is.